bootstrap.sh

```diff
@@ -0,0 +1,175 @@
+#! /bin/bash
+banner() {
+	echo "|------------------------------------------|"
+	echo "|----- Welcome to the redox bootstrap -----|"
+	echo "|------------------------------------------|"
+}
+
+osx()
+{
+	echo "Detected OSX!"
+	temp=$(brew --version)
+	if [ -n "$temp" ]; then
+		echo "Homebrew detected! Now updating..."
+		brew update
+		echo "Now installing git..."
+		brew install git
+		echo "Now installing virtualbox..."
+		brew cask install virtualbox
+	else
+		echo "Homebrew does not appear to be installed! Would you like me to install it?"
+		printf "(Y/n): "
+		#For now assume yes
+		ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+	fi
+	echo "Cloning Redox repo"
+	git clone -b $1 --recursive https://github.com/redox-os/redox.git
+	sh redox/setup/osx-homebrew.sh
+	sh redox/setup/binary.sh
+}
+
+archLinux()
+{
+	echo "Detected Arch Linux, mah fav"
+	echo "Updating system..."
+	sudo pacman -Syu
+	echo "Installing git..."
+	sudo pacman -S git
+	echo "Cloning Redox repo"
+    if [ "$2" == "qemu" ]; then
+        echo "Installing QEMU..."
+        sudo pacman -S qemu
+    fi
+	git clone -b $1 --recursive https://github.com/redox-os/redox.git
+	sh redox/setup/arch.sh
+	sh redox/setup/binary.sh
+}
+
+ubuntu()
+{
+	echo "Detected Ubuntu/Debian"
+	echo "Updating system..."
+	sudo apt-get upgrade
```

Comment:
Since an upgrade of the system is not really necessary to get up and
running redox I think, drop it. Still we need to update package repositorys
via `apt-get update` to avoid problems when installing later. If you don't
do it, you might run into troubles because package versions shifted since
last update.

DomThePorcupine <notifications@github.com> schrieb am Do., 17. Dez. 2015 um
08:01 Uhr:

> In bootstrap.sh
> <https://github.com/redox-os/redox/pull/383#discussion_r47875060>:
>
> > +	sudo pacman -S git
> > +	echo "Cloning Redox repo"
> > +    if [ "$2" == "qemu" ]; then
> > +        echo "Installing QEMU..."
> > +        sudo pacman -S qemu
> > +    fi
> > +	git clone -b $1 --recursive https://github.com/redox-os/redox.git
> > +	sh redox/setup/arch.sh
> > +	sh redox/setup/binary.sh
> > +}
> > +
> > +ubuntu()
> > +{
> > +	echo "Detected Ubuntu/Debian"
> > +	echo "Updating system..."
> > +	sudo apt-get upgrade
>
> So would you suggest removing the update part completely?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/redox-os/redox/pull/383/files#r47875060>.
>